File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Controllers/Controller.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LabAssig7.Repository;
using LabAssig7.Model;
using LabAssig7.Model.ADT;
using LabAssig7.Model.Statements;
namespace LabAssig7.Controllers
{
    public class Controller
    {
        private IPrgRepository repo;

        public Controller(IPrgRepository r)
        {
            this.repo = r;
        }

        public void executeOneStep()
        {
            PrgState p = repo.getCurrentPrgState();
            Statement stmt = p.Stack.pop();

            stmt.Execute(p);
            Console.WriteLine(p);
        }

        public void executeAllSteps()
        {
            PrgState p = repo.getCurrentPrgState();
            while(!p.Stack.IsEmpty())
            {
                this.executeOneStep();
                repo.logPrgState();
            }
        }

    }
}

File: MAP(Advanced Programming Methods)/lab2v2/src/Controller/Controller.java
package Controller;
import Model.IStack;
import Model.PrgState;
import Model.Statement;
import Repository.PrgRepository;
import Repository.IRepository;

public class Controller {
    IRepository repo;

    public Controller(IRepository r)
    {
        this.repo=r;
    }

    public void executeOneStep()
    {
        PrgState prg= repo.getCurrentPrgState();
        IStack<Statement> exestack=prg.getExeStack();
        if(exestack.isEmpty())
            return;
        Statement instruction = exestack.pop();
        instruction.execute(prg);
        System.out.println(prg);
    }

    public void executeAll()
    {
        PrgState ps = repo.getCurrentPrgState();
        IStack<Statement> exeStack = ps.getExeStack();
        while(!exeStack.isEmpty())
        {
            this.executeOneStep();
        }
    }
}
